Doesn't work with FetchContent

When I try to use the library with FetchContent:

```cmake
include(FetchContent)

FetchContent_Declare(
        reduct_cpp
        URL https://github.com/reduct-storage/reduct-cpp/archive/refs/tags/v0.6.0.zip
        URL_HASH MD5=8be8b5154193e451fc1599b9ecfeb701
)

FetchContent_MakeAvailable(reduct_cpp)
```

I have this error:
```
ERROR: Conanfile not found at /home/atimin/Projects/flipback/reduct-storage/cli-client/conanfile.py
CMake Error at cmake-build-debug/conan.cmake:631 (message):
  Conan install failed='1'
Call Stack (most recent call first):
  cmake-build-debug/_deps/reduct_cpp-src/cmake/InstallDependencies.cmake:14 (conan_cmake_install)
  cmake-build-debug/_deps/reduct_cpp-src/CMakeLists.txt:19 (include)

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,8 @@
-## Release 0.7.0 (in progress)
+## [Unreleased]
+
+### Fixed
+
+- Cmake Fetch_Content integration, [PR-33](https://github.com/reduct-storage/reduct-cpp/pull/33)
 
 ## Release 0.6.0 (2022-06-29)
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,10 +21,6 @@ include(cmake/InstallDependencies.cmake)
 configure_file(cmake/ReductCppConfig.cmake.in ${CMAKE_BINARY_DIR}/cmake/ReductCppConfig.cmake @ONLY)
 configure_file(cmake/ReductCppConfigVersion.cmake.in ${CMAKE_BINARY_DIR}/cmake/ReductCppConfigVersion.cmake @ONLY)
 
-
-include_directories(${CMAKE_SOURCE_DIR}/src)
-include_directories(${CMAKE_SOURCE_DIR}/3dparty)
-
 add_subdirectory(src)
 if (REDUCT_CPP_ENABLE_TESTS)
     add_subdirectory(tests)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,7 @@ project(reductcpp VERSION ${REDUCT_CPP_FULL_VERSION})
 
 message(STATUS "Version ${REDUCT_CPP_FULL_VERSION}")
 
+set(REDUCT_CPP_INSTALL_DEPS ON CACHE BOOL "Install dependencies by using conan or fetch content")
 set(REDUCT_CPP_ENABLE_TESTS ON CACHE BOOL "Compile tests")
 set(REDUCT_CPP_ENABLE_EXAMPLES ON CACHE BOOL "Compile examples")
 
@@ -16,7 +17,9 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
-include(cmake/InstallDependencies.cmake)
+if (REDUCT_CPP_INSTALL_DEPS)
+    include(cmake/InstallDependencies.cmake)
+endif()
 
 configure_file(cmake/ReductCppConfig.cmake.in ${CMAKE_BINARY_DIR}/cmake/ReductCppConfig.cmake @ONLY)
 configure_file(cmake/ReductCppConfigVersion.cmake.in ${CMAKE_BINARY_DIR}/cmake/ReductCppConfigVersion.cmake @ONLY)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,6 @@ project(reductcpp VERSION ${REDUCT_CPP_FULL_VERSION})
 
 message(STATUS "Version ${REDUCT_CPP_FULL_VERSION}")
 
-set(REDUCT_CPP_INSTALL_DEPS ON CACHE BOOL "Install dependencies by using conan or fetch content")
 set(REDUCT_CPP_ENABLE_TESTS ON CACHE BOOL "Compile tests")
 set(REDUCT_CPP_ENABLE_EXAMPLES ON CACHE BOOL "Compile examples")
 
@@ -17,9 +16,7 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
-if (REDUCT_CPP_INSTALL_DEPS)
-    include(cmake/InstallDependencies.cmake)
-endif()
+include(cmake/InstallDependencies.cmake)
 
 configure_file(cmake/ReductCppConfig.cmake.in ${CMAKE_BINARY_DIR}/cmake/ReductCppConfig.cmake @ONLY)
 configure_file(cmake/ReductCppConfigVersion.cmake.in ${CMAKE_BINARY_DIR}/cmake/ReductCppConfigVersion.cmake @ONLY)

--- a/cmake/InstallDependencies.cmake
+++ b/cmake/InstallDependencies.cmake
@@ -11,7 +11,7 @@ include(${CMAKE_BINARY_DIR}/conan.cmake)
 find_program(CONAN_CMD conan)
 if (CONAN_CMD)
     conan_cmake_autodetect(settings)
-    conan_cmake_install(PATH_OR_REFERENCE ${CMAKE_SOURCE_DIR}/conanfile.py
+    conan_cmake_install(PATH_OR_REFERENCE ${CMAKE_CURRENT_SOURCE_DIR}/conanfile.py
             BUILD missing
             SETTINGS ${settings})
 

--- a/conanfile.py
+++ b/conanfile.py
@@ -18,7 +18,7 @@ class DriftFrameworkConan(ConanFile):
                 "cpp-httplib/0.10.8",
                 "nlohmann_json/3.10.5",
                 "openssl/3.0.3",
-                "catch2/2.13.9")
+                "catch2/2.13.8")
 
     def config_options(self):
         if self.settings.os == "Windows":

--- a/conanfile.py
+++ b/conanfile.py
@@ -18,7 +18,7 @@ class DriftFrameworkConan(ConanFile):
                 "cpp-httplib/0.10.8",
                 "nlohmann_json/3.10.5",
                 "openssl/3.0.3",
-                "catch2/2.13.8")
+                "catch2/2.13.9")
 
     def config_options(self):
         if self.settings.os == "Windows":

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -4,9 +4,6 @@ set(SRC_FILES
         reduct/bucket.cc
         reduct/client.cc
         reduct/error.cc
-        ../3dparty/sha256.h
-        reduct/internal/http_client.h
-        reduct/internal/serialisation.h
         )
 
 set(PUBLIC_HEADERS

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -23,7 +23,7 @@ target_include_directories(${CMAKE_PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_
 
 target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE dependencies)
 set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES PUBLIC_HEADER "${PUBLIC_HEADERS}")
-
+target_compile_definitions(${CMAKE_PROJECT_NAME} PUBLIC CPPHTTPLIB_OPENSSL_SUPPORT)
 
 include(GNUInstallDirs)
 install(TARGETS ${CMAKE_PROJECT_NAME}

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -3,7 +3,11 @@ set(SRC_FILES
         reduct/internal/serialisation.cc
         reduct/bucket.cc
         reduct/client.cc
-        reduct/error.cc)
+        reduct/error.cc
+        ../3dparty/sha256.h
+        reduct/internal/http_client.h
+        reduct/internal/serialisation.h
+        )
 
 set(PUBLIC_HEADERS
         reduct/bucket.h
@@ -12,9 +16,14 @@ set(PUBLIC_HEADERS
         reduct/http_options.h
         reduct/result.h)
 
+
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
 
 add_library(${CMAKE_PROJECT_NAME} STATIC ${SRC_FILES})
+
+target_include_directories(${CMAKE_PROJECT_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+target_include_directories(${CMAKE_PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../3dparty)
+
 target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE dependencies)
 set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES PUBLIC_HEADER "${PUBLIC_HEADERS}")
 

